packages/components/src/popover/index.js

```diff
@@ -361,38 +219,109 @@ const Popover = ( {
 } ) => {
 	const anchorRefFallback = useRef( null );
 	const contentRef = useRef( null );
+	const containerRef = useRef();
+	const [ isMobileViewport, setIsMobileViewport ] = useState( false );
```

Comment:
This should help here to avoid an Adhoc solution https://github.com/WordPress/gutenberg/pull/18816